### PR TITLE
Calculator

### PR DESCRIPTION
**Pull request by Stanislava88**
Here i have solved the task for calculator that calculate whole numbers

http://stanislava88.github.io/calculator/

### CODE REVIEW

File: src/com/clouway/calculator/calculator.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+function calculate() {
+    var result;
+    var field = document.getElementById('display');
+    var lastElement = field.value.charAt(field.value.length - 1);
+
+    if (!isEmpty(field.value) && isDigit(lastElement)) {
+        result = eval(field.value);
+        
+        if (result == Infinity) {
```
Comment (stefan-dimitrov) [inline src/com/clouway/calculator/calculator.js]: You should also check for negative infinity. 
Result of `1 - 3 / 0`:

screenshot from 2016-06-06 08 57 09

Comment (Stanislava88) [inline src/com/clouway/calculator/calculator.js]: done

**Comment by Stanislava88** [comment]
Added check for negative infinity

File: src/com/clouway/calculator/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <script src="calculator.js"></script>
+    <link rel="stylesheet" href="style.css">
+</head>
+<body>
+<div id="calculator">
+    <div class="keys">
+
+        <input type="text" class="display" id="display" disabled>
+
+        <button id="reset" onclick="addValue('clear')">Clear</button>
```
Comment (freekman) [inline src/com/clouway/calculator/index.html]: this is strange ```addValue('clear')``` you should have a clear function or something

File: src/com/clouway/calculator/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        <button id="cl" type="button" onclick="clearLast()">←</button>
+        <br><br>
```
Comment (freekman) [inline src/com/clouway/calculator/index.html]: ?

**Comment by freekman** [comment]
i dont find this to be accurate 
screenshot from 2016-06-06 10 52 10

**Comment by freekman** [comment]
try to devide 99999/3 tell me what is wrong

**Comment by Stanislava88** [comment]
this is because i have  check```!isNan(result)``` . I fixed this using check ``!isFinite(result)```

File: src/com/clouway/calculator/calculator.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/**
+ * Created by Stanislava on 01.06.16.
+ */
+function addValue(button) {
```
Comment (freekman) [inline src/com/clouway/calculator/calculator.js]: i think it would be better if you use different functions to ```addNumber() and addSign()```

**Comment by freekman** [comment]
when you have fixed the comments write me a notification to review the PR again

File: src/com/clouway/calculator/index.html (hunk 2 above)
Comment (Stanislava88) [inline src/com/clouway/calculator/index.html]: done

**Comment by Stanislava88** [comment]
done

File: src/com/clouway/calculator/calculator.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            field.value = "Can not divide by zero";
+        } else {
+            field.value = result;
+        }
+    } else {
+        field.value = "Error";
+    }
+}
+
+function clearAll() {
```
Comment (freekman) [inline src/com/clouway/calculator/calculator.js]: just clear() is OK

File: src/com/clouway/calculator/style.css [hunk 6]
```diff
@@ -0,0 +1,90 @@
+* {
```
Comment (freekman) [inline src/com/clouway/calculator/style.css]: leave it like it is but just for you to know it is a bad practice to do this
```js
*{
..................
}
```

if you want to do this properly you should use a wrapper EX:
```js
<body>
  <div class="wrapper">
  ..........................
  </div>
</body>
```

File: src/com/clouway/calculator/calculator.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+function isEmpty(field) {
+    if (field == '') {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+function isDigit(button) {
+    if (button != '/' && button != '*' && button != '+' && button != '-') {
```
Comment (freekman) [inline src/com/clouway/calculator/calculator.js]: i think if this is done using regex it would look better

**Comment by freekman** [comment]
LGTM

**Comment by stefan-dimitrov** [comment]
LGTM.

**Pull request merged**